### ISSUE TITLE
Post processing shaders

### ISSUE THREAD

**Issue by nukeop** [LIKO-12/LIKO-12#138]
Idea: allow developers to apply their own fullscreen post-processing shaders

**Comment by pablomayobre** [comment]
Do we want post processing only or do we also need per-object shaders?
* I want to apply a shader to the final rendered frame only
* I render an image and I want to apply a 3D transformation in a shader, to that specific image

Can we apply the pixelation and palette reduction shader at will, for example, we apply a post-processing shader that does some kind of blur but we want the effect to still be pixelated and limited to the palette... do I need to rewrite LIKO's shader?

* Aesthetic finishing, say a CRT shader or similar
```
LIKO shader
Custom post processing shader
```

* Transitional effect
```
LIKO shader
Custom post processing shader
LIKO shader
```

* I cant think of a good case for this but:
```
Custom post processing shader
LIKO shader
```

EDIT: This comment is in order to clarify what will be implemented and why the other approaches won't, not necessarily requesting that all this features make it in.

LIKO-12 as a fantasy console offers strict limitations, and that makes it cooler! Enabling all these features would probably result in too much freedom for a FC

**Comment by Rami-Sabbagh** [comment]
Hey, I talked with him using Discord, and he wants post shaders that are active when rendering the canvas, to do some effects like CRT.

**Comment by Rami-Sabbagh** [comment]
I'll have to create another canvas in-order to achieve this.

**Comment by pablomayobre** [comment]
Yes such approach is generally called a backbuffer, you can generally reuse canvases when doing many passes

**Comment by nukeop** [comment]
Do you see this as more of a DiskOS feature, or would it be applied per game?

**Comment by Rami-Sabbagh** [comment]
No, I won't allow games to write their own shaders.

I will create a `Shaders` folder in the appdata folder, where you can place your own shaders.

And cycle between them somehow using a hotkey.
I may allow games to set the active shader, but not create new ones.

**Comment by Rami-Sabbagh** [comment]
Implemented in fa8819a421c197563bcea5f0bb2a2ef9bf38f9f0
